### PR TITLE
Add apartment form repository

### PR DESCRIPTION
**Pull request by koral--**
Fix accommodation tests

### CODE REVIEW

**Review by Pablo1785 — CHANGES_REQUESTED** [review]
Please rename Apartment to Accomodation in new class and file names to keep things consistent

File: lib/presentation/host/widgets/apartment_form.dart [hunk 1]
```diff
@@ -32,7 +32,11 @@ class _ApartmentFormState extends State<ApartmentForm> {
   @override
   Widget build(BuildContext context) => BlocListener<ApartmentFormBloc, ApartmentFormState>(
         listener: (BuildContext context, state) {
-          //TODO
+          if (state == const ApartmentFormState.submitSuccess()) {
+            AutoRouter.of(context).push(const ApartmentAddedSuccessRoute());
+          } else if (state is SubmitFailure) {
```
Comment (Pablo1785) [inline lib/presentation/host/widgets/apartment_form.dart]: You can use state.maybeMap(submitSuccess: (state) {}, someDifferentApartmentFormState: (state){}, otherOneOfMyApartmentFormStates: (state) {}, orElse: () {}) to handle select ApartmentFormStates without making SubmitFailure a public type

**Comment by koral--** [comment]
done

File: lib/application/apartment_form/apartment_form_state.dart [hunk 2]
```diff
@@ -9,5 +9,5 @@ class ApartmentFormState with _$ApartmentFormState {
 
   const factory ApartmentFormState.submitSuccess() = _SubmitSuccess;
 
-  const factory ApartmentFormState.submitFailure(Failure failure) = _SubmitFailure;
+  const factory ApartmentFormState.submitFailure(Failure failure) = SubmitFailure;
```
Comment (Pablo1785) [inline lib/application/apartment_form/apartment_form_state.dart]: Should be _SubmitFailure and _SubmitFailure should not be used directly anywhere else, instead use state.map() wherever you need to check all possible states and state.maybeMap() wherever you only need to check for some of the states

File: lib/application/accommodation_form/accommodation_form_state.dart [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+part 'accommodation_form_state.freezed.dart';
+
+@freezed
+class AccommodationFormState with _$AccommodationFormState {
+  const factory AccommodationFormState.initial() = _Initial;
+
+  const factory AccommodationFormState.submitSuccess() = _SubmitSuccess;
+
+  const factory AccommodationFormState.submitFailure(Failure failure) = SubmitFailure;
```
Comment (Pablo1785) [inline lib/application/accommodation_form/accommodation_form_state.dart]: Should be _SubmitFailure and shouldnt be used outside of this definition

File: test/application/accomodation_loader/accomodation_loader_bloc_test.dart [hunk 4]
```diff
@@ -28,7 +28,7 @@ void main() {
       // assert
       expect(
         accomodationLoaderBloc.state,
-        const AccomodationLoaderState.initial(),
+        const AccommodationLoaderState.initial(),
```
Comment (Pablo1785) [inline test/application/accomodation_loader/accomodation_loader_bloc_test.dart:30]: The bloc is renamed here but there are no changes to its definition, this should be all red in the IDE/tests

Please revert changes to this file or rename the AccomodationLoader to AccommodationLoader as well

**Review by Pablo1785 — CHANGES_REQUESTED** [review]

**Pull request merged**